x-pack/plugins/security_solution/public/timelines/components/timeline/data_providers/provider_item_actions.tsx

```diff
@@ -121,14 +136,26 @@ export const getProviderActions = ({
         name: i18n.FILTER_FOR_FIELD_PRESENT,
         onClick: onFilterForFieldPresent,
       },
+      timelineType === TimelineType.template
+        ? {
+            className: CONVERT_TO_FIELD_CLASS_NAME,
+            disabled: isLoading,
+            icon: 'visText',
+            name:
+              type === DataProviderType.template
+                ? i18n.CONVERT_TO_FIELD
+                : i18n.CONVERT_TO_TEMPLATE_FIELD,
+            onClick: toggleType,
+          }
+        : null,
       {
         className: DELETE_CLASS_NAME,
         disabled: isLoading,
         icon: 'trash',
         name: i18n.DELETE_DATA_PROVIDER,
         onClick: deleteItem,
       },
-    ],
+    ].filter((item) => item != null) as EuiContextMenuPanelItemDescriptor[],
```

Comment:
To avoid the `as`, you can do that
```
{
    id: 0,
    items: [
      {
        className: EDIT_CLASS_NAME,
        disabled: isLoading,
        icon: 'pencil',
        name: i18n.EDIT_MENU_ITEM,
        panel: 1,
      },
      {
        className: EXCLUDE_CLASS_NAME,
        disabled: isLoading,
        icon: `${isExcluded ? 'plusInCircle' : 'minusInCircle'}`,
        name: isExcluded ? i18n.INCLUDE_DATA_PROVIDER : i18n.EXCLUDE_DATA_PROVIDER,
        onClick: toggleExcluded,
      },
      {
        className: ENABLE_CLASS_NAME,
        disabled: isLoading,
        icon: `${isEnabled ? 'eyeClosed' : 'eye'}`,
        name: isEnabled ? i18n.TEMPORARILY_DISABLE_DATA_PROVIDER : i18n.RE_ENABLE_DATA_PROVIDER,
        onClick: toggleEnabled,
      },
      {
        className: FILTER_FOR_FIELD_PRESENT_CLASS_NAME,
        disabled: isLoading,
        icon: 'logstashFilter',
        name: i18n.FILTER_FOR_FIELD_PRESENT,
        onClick: onFilterForFieldPresent,
      },
      timelineType === TimelineType.template
        ? {
            className: CONVERT_TO_FIELD_CLASS_NAME,
            disabled: isLoading,
            icon: 'visText',
            name: (
              <>
                {type === DataProviderType.template
                  ? i18n.CONVERT_TO_FIELD
                  : i18n.CONVERT_TO_TEMPLATE_FIELD}
              </>
            ),
            onClick: toggleType,
          }
        : { name: null },
      {
        className: DELETE_CLASS_NAME,
        disabled: isLoading,
        icon: 'trash',
        name: i18n.DELETE_DATA_PROVIDER,
        onClick: deleteItem,
      },
    ].filter((item) => item.name != null),
  },
```